docs/stratum.md

```diff
@@ -0,0 +1,123 @@
+For the following jobs being sent to the miner, a recalculation of the difficulty is in order. If the difference between new diffuculty and previous difficulty is larger than `share_target_diff_threshold` (in percents), new difficulty is used for next jobs sent to miner, via sending `set_target` notification to miner.
+If the difference is smaller than the threshold, previous difficulty is used for the following jobs.
+(default 5.0)
+
+`edge_bits` - (optional) - a proof of work algorithm we use is called Cuckoo Cycle - looks for a cycle in randomly generated graph. `edge_bits` determines how long this cycle should be.
+You should only change this value if you plan to deploy your own network, as for this change to work, you will need to distribute a new mining client.
+(default 29)
+
+`max_jobs` - (optional) - max number of jobs kept in queue for each client. Lower number of `max_jobs` results in more frequent changes via `set_target` notification.
+(default 20)
+
+`max_workers` - (optional) - max number of workers for one connection. This is not fully implemented yet, will be required for a proxy implementation.
+(default 20)
+
+`msg_timeout` - (optional) - timeout (in milliseconds) guarding transitions between miner states (connected, configured, subscribed). If a transition doesn't happen in time, the client session is closed.
+(default 15)
+
+
+REWARD section
+==========
+
+`reward_last_rounds` - (optional) - how many epochs between keyblocks we reward from a received mining reward. Higher number of epochs would cause that the mining reward will be distributed to more shares, but each partial reward will be smaller. Higher number supports loayality of the miners, by raising a chance of rewarding miners who contributed shares for a longer period of time.
+(default 2)
+
+`beneficiaries` - a list in "account_address:percent_share" format - denoting the pool operator addresses with their respective shares. The sum of shares must be less than or equal to 100.
+The sum of shares equal to 100 effectively means that all rewards coming from mining are distributed to pool operators, while miners don't receive any reward.
+
+`keys -> dir` - points to a location where to store the stratum account keypair. Since we want to send tokens to miners from this account, we need to have access to its private key.
+The stratum account holds reward(s) from mining which are scheduled to be payed out via a payment contract. While it may sound tempting, stratum account should never have any significant accumulated wealth since it only serves as a intermittent place for receiving of reward, out of which are then the funds distrubuted to miners soon.
+If `dir` represents a relative path, the directory will be in `priv` directory of aestratum app. Providing absolute path allows storing of keypair outside of the aestratum app.
```

Comment:
It's worth mentioning that this keys are of intermediate account and should be backed up, otherwise some reward (180 + 2 (default)) can be lost in case of lost keys because of server/disk outage?

Reply:
added